src/errors/runtime/index.js

```diff
@@ -74,13 +95,24 @@ export class APIError extends Error {
 
         return this.message + ERROR_SEPARATOR + renderedCallsite;
     }
+
+    static _prepareTemplateAndArgsIfNecessary (template, args) {
+        const lastArg = args.pop();
+
+        if (lastArg instanceof ProcessTemplateInstruction)
+            template = lastArg.processFn(template);
+        else
+            args.push(lastArg);
+
+        return template;
+    }
 }
 
 export class ClientFunctionAPIError extends APIError {
-    constructor (methodName, instantiationCallsiteName, template, ...args) {
-        template = template.replace(/\{#instantiationCallsiteName\}/g, instantiationCallsiteName);
+    constructor (methodName, instantiationCallsiteName, code, ...args) {
+        args.push(new ProcessTemplateInstruction(template => template.replace(/\{#instantiationCallsiteName\}/g, instantiationCallsiteName)));
```

Comment:
I have been trying to find a good solution to get rid of unclear operations with adding/removing something in `args` and using extra class `ProcessTemplateInstruction`. I couldn't find something flawless, but maybe the following approach would be more clear:
```js
export class APIError extends Error {
    constructor (methodName, codeOrTemplate, ...args) {
        const template = TEMPLATES[codeOrTemplate] || codeOrTemplate; 
        //OR const template = RUNTIME_ERRORS.includes(codeOrTemplate) ? TEMPLATES[codeOrTemplate] : codeOrTemplate;

        const rawMessage = renderTemplate(template, ...args);
        //....
    }
}

export class ClientFunctionAPIError extends APIError {
    constructor (methodName, instantiationCallsiteName, code, ...args) {
        const template = TEMPLATES[code].replace(/\{#instantiationCallsiteName\}/g, instantiationCallsiteName);

        super(methodName, template, ...args);
    }
}
```
In this case we will not have to change `args`, use `_prepareTemplateAndArgsIfNecessary` and `ProcessTemplateInstruction`

Reply:
Yep, we still need pass `code ` to `APIError`. Oook,  let's stay as it is